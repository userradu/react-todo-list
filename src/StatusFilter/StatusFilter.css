.filter-option {
  background-color: transparent;
  border: none;
  color: #428BCA;
}

.filter-option:hover {
  cursor: pointer;
}

.filter-option.selected {
  background-color: #428BCA;;
  color: #fff;
}
