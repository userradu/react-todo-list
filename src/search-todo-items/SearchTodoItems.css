.status-filter {
  margin-left: 10px;
}
