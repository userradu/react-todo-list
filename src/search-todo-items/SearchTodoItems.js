import React from 'react';

class SearchTodoItems extends React.Component {
    constructor(props) {
      super(props);
      this.state = {searchedText: ''};
      this.handleSearchInputChange = this.handleSearchInputChange.bind(this);
    }

    handleSearchInputChange(event) {
      const searchedText = event.target.value;
      this.setState({
        searchedText: searchedText
      });
      this.props.onSearchInputChange(searchedText);
    }

    render() {
      return (
        <div>
          <input
            type="text"
            placeholder="Search"
            value={this.state.searchedText}
            className={this.props.className}
            onChange={this.handleSearchInputChange}/>
        </div>
      );
    }
}

export default SearchTodoItems;
