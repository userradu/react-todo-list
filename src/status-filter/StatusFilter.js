import React from 'react';
import './StatusFilter.css';

function StatusFilter(props) {
  return (
    <span className={props.className}>
      <button
        className={"filter-option" + (props.currentStatus === props.status ? ' selected' : '')}
        onClick={() => props.onStatusFilterChanged(props.status)}>
        {props.status}
      </button>
    </span>
  );
}

export default StatusFilter;
