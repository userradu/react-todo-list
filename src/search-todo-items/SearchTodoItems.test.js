import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import SearchTodoItems from './SearchTodoItems';
import StatusFilter from '../status-filter/StatusFilter';

describe('SearchTodoItems', () => {

  it('renders correctly', () => {
    const tree = renderer
      .create(<SearchTodoItems
        onSearchInputChange={() => { }}
        statusFilter="All"
        className="className"
        onStatusFilterChanged={() => { }} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('sets the state with the updated input value', () => {
    const searchValue = 'search value';

    const wrapper = shallow(<SearchTodoItems
      onSearchInputChange={() => { }}
      statusFilter="All"
      className="className"
      onStatusFilterChanged={() => { }} />);

    expect(wrapper.state().searchedText).toEqual('');

    const input = wrapper.find('input');
    input.simulate('change', {
      target: { value: searchValue }
    });

    expect(wrapper.state().searchedText).toEqual(searchValue);
  })

  it('calls props.onSearchInputChange() with the correct arguments', () => {
    const onSearchInputChange = sinon.spy();
    const searchValue = 'search value';

    const wrapper = shallow(<SearchTodoItems
      onSearchInputChange={onSearchInputChange}
      statusFilter="All"
      className="className"
      onStatusFilterChanged={() => { }} />);

    const input = wrapper.find('input');
    input.simulate('change', {
      target: { value: searchValue }
    });
    expect(onSearchInputChange.called).toBe(true);
    expect(onSearchInputChange.calledWith(searchValue)).toBe(true);
  });

  it('creates a StatusFilter component for each status filter option', () => {
    const wrapper = shallow(<SearchTodoItems
      onSearchInputChange={() => { }}
      statusFilter="All"
      className="className"
      onStatusFilterChanged={() => { }} />);

    expect(wrapper.find(StatusFilter)).toHaveLength(wrapper.instance().statusFilterOptions.length);
  });

});
