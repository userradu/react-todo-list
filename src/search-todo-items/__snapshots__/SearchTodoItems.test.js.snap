// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchTodoItems renders correctly 1`] = `
<div>
  <input
    className="className"
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
  <span
    className="status-filter"
  >
    <button
      className="filter-option selected"
      onClick={[Function]}
    >
      All
    </button>
  </span>
  <span
    className="status-filter"
  >
    <button
      className="filter-option"
      onClick={[Function]}
    >
      Incomplete
    </button>
  </span>
  <span
    className="status-filter"
  >
    <button
      className="filter-option"
      onClick={[Function]}
    >
      Complete
    </button>
  </span>
</div>
`;
